source/controlTypes/state.py

```diff
@@ -0,0 +1,181 @@
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2007-2021 NV Access Limited, Babbage B.V.
+
+from enum import IntEnum
+from typing import Dict
+
+from utils.displayString import DisplayStringEnumMixin, DisplayStringEnumMixinMeta
+
+
+class State(DisplayStringEnumMixin, IntEnum, metaclass=DisplayStringEnumMixinMeta):
+	@property
+	def _displayStringLabels(self):
+		return _stateLabels
+
+	@property
+	def defaultValue(self):
+		return self.DEFUNCT

```

Comment:
I was just trying to fill in default values that hopefully made some sense. On `DisplayStringEnumMixin` I can turn `defaultValue` into a `property` that returns `None` rather than an `abstractproperty` and then reraise the same `KeyError` that would be raised if there was no matching label.

Reply:
The danger is that it leads developers to use that value whenever they aren't sure, rather than really consider what should be used. I'd like to remove the default value from the base class, it isn't clear that ALL concrete classes will have a default value and therefore breaks the "is-a" contract between base and child class. See [Liskov substitution principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle).

If there are any enums that have a confident default value, those specializations can extend the interface.